src/torchmetrics/segmentation/mean_iou.py

```diff
@@ -120,10 +121,11 @@ def update(self, preds: Tensor, target: Tensor) -> None:
         )
         score = _mean_iou_compute(intersection, union, per_class=self.per_class)
         self.score += score.mean(0) if self.per_class else score.mean()
+        self.num_batches += 1
 
     def compute(self) -> Tensor:
-        """Update the state with the new data."""
-        return self.score  # / self.num_batches
```

Comment:
@SkafteNicki any idea why this was commented?

Reply:
I think the best would be to add a testcase to `MetricTester`, that checks if the return of `self.compute` is the same, when class is updated using `self.forward` and `self.update`